specifications/todo-api.yaml

```diff
@@ -0,0 +1,99 @@
+openapi: 3.0.2
+
+info:
+  title: TODO API
+  description: |-
+    This API is a backend for the TODO example
+  version: 0.1.0
+
+paths:
+  /todo-items:
+    get:
+      operationId: list-todo-items
+      description: Lists all TODO items
+      responses:
+        "200":
+          description: Ok
+          content:
+            "application/json":
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/todo-item"
+
+    post:
+      operationId: add-todo-item
+      description: Add a todo-item
+      requestBody:
+        required: true
+        content:
+          "application/json":
+            schema:
+              $ref: "#/components/schemas/todo-item"
```

Comment:
We need a different schema to create a todo item. It probably won't have an idea and a "done" field.

Reply:
I don't understand what you mean sir

Reply:
So if we get (query) the todo items we want a list with items that have an id, description and done field. One of those items looks something like this:

```yaml
id: 1
description: I need to do some laundry
done: false
```

This is then quite easy to put in a list.

But!

If we want to create a todo item, then we want only the description field! So, this will look like this

```yaml
description: Fold laundry
```

If this item is created, then the id field and done field are automatically filled. The id is a generated identity, usually a number. The done field is false, because if we create a todo item, then it is not done.

So the model of the toto item that we retrieve (query) is a different todo item that we create (command). This should remind you of the CQRS principle that query (retrieval, select, etc) is seperate from command (insert, update, delete, etc).

Reply:
Yes, I understand now! Thank you